src/test/kotlin/com/xenomachina/argparser/ArgParserTest.kt

```diff
@@ -421,6 +421,24 @@ class DefaultWithProviderTest : Test({
     Args(parserOf("-x9", "-x8")).x shouldBe 8
 })
 
+class DefaultWithLambda : Test({
+
+    var a = 1
```

Comment:
This is a bit hard to follow. Why not have `var defaultCalled = false` inside `Args`? `defaultCalled shouldBe false` is more self-documenting than `a shouldBe 1`.